src/main.ts

```diff
@@ -272,6 +263,23 @@ $(function(){
     eagle.errorsMode(Errors.Mode.Graph);
 });
 
+async function loadRepos() {
+    // Get the list of git repos
+    if (UiModeSystem.getActiveUiMode().getName()==='Student'){
```

Comment:
The call to autoLoad should take place even if repositories fail to load.

Actually, there is no way for the repository load to fail. Even if an error occurs, we can always just return an empty list. So I've removed the 'reject' case from the async function, it will always resolve.